packages/concerto-core/lib/decoratorutil.js

```diff
@@ -0,0 +1,371 @@
+    static constructDCSDictionary(dictionary, key, value, options) {
+        const val = {
+            declaration:options?.declaration || '',
+            property:options?.property || '',
+            mapElement:options?.mapElement || '',
+            dcs: JSON.stringify(value),
+        };
+        if (dictionary[key] && Array.isArray( dictionary[key])) {
+            dictionary[key].push(val);
+        } else {
+            dictionary[key] = [val];
+        }
+        return dictionary;
+    }
+    /**
+    * parses the extracted decorators and generates arrays of decorator command set and vocabularies
+    *
+    * @param {Object} extractionDictionary - extracted decorators and vocabularies
+    * @param {String} DCS_VERSION - dcs_version
+    * @param {String} locale - locale for vocabularies
+    * @returns {Object} - constructed DCS Dict and processed models ast
+    * @private
+    */
+    static parseDecosAndVocabs(extractionDictionary, DCS_VERSION, locale){
+        const decoratorData = [];
+        const vocabData = [];
+        Object.keys(extractionDictionary).forEach(namespace => {
+            const {name, version} = ModelUtil.parseNamespace(namespace);
+            const nameOfDcs = name;
+            const versionOfDcs = version;
+            const jsonData = extractionDictionary[namespace];
+            const patternToDetermineVocab = /^Term_/i;
+            let dcsObjects = [];
+            let vocabObject = {};
+            jsonData.forEach(obj =>{
+                const decos = JSON.parse(obj.dcs);
+                const target = {
+                    '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.CommandTarget`,
+                    'namespace':namespace
+                };
+                if (obj.declaration && obj.declaration!==''){
+                    target.declaration = obj.declaration;
+                }
+                if (obj.property && obj.property!==''){
+                    target.property = obj.property;
+                }
+                if (obj.mapElement && obj.mapElement!==''){
+                    target.mapElement = obj.mapElement;
+                }
+                decos.forEach(dcs =>{
+                    if (dcs.name !== 'Term' && !patternToDetermineVocab.test(dcs.name)){
+                        dcsObjects = this.parseNonVocabularyDecorators(dcsObjects, dcs, DCS_VERSION, target);
+                    }
+                    else{
+                        vocabObject = this.parseVocabularies(vocabObject, obj, dcs);
+                    }
+                });
+            });
+            if (dcsObjects?.length > 0){
+                const dcmsForNamespace = {
+                    '$class': `org.accordproject.decoratorcommands@${DCS_VERSION}.DecoratorCommandSet`,
+                    'name': nameOfDcs,
+                    'version': versionOfDcs,
+                    'commands': dcsObjects
+                };
+                decoratorData.push(dcmsForNamespace);
+            }
```

Comment:
you can now create another function that "transform"s `dcsObjects`, maybe called `transformNonVoabularyDecorators`. I know this is a simple transformation, but it would help to contain transformations in one place. Then you can return the `dcmsForNamespace` and push that to `decoratorData`.